xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp

```diff
@@ -140,12 +143,23 @@ CRendererBase::CRendererBase(CVideoSettings& videoSettings)
 
 CRendererBase::~CRendererBase()
 {
-  if (DX::Windowing()->IsHDROutput())
+  if (m_AutoSwitchHDR)
   {
-    CLog::LogF(LOGDEBUG, "Restoring SDR rendering");
-    DX::Windowing()->SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709);
-    if (m_AutoSwitchHDR)
-      DX::Windowing()->ToggleHDR(); // Toggle display HDR OFF
+    if (m_isHdrEnabled)
+    {
+      if (!DX::Windowing()->IsHDROutput())
+	  {
+	    CLog::LogF(LOGINFO, "Restoring HDR rendering (AutoSwitch)");
+        DX::Windowing()->SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020);
+	    DX::Windowing()->ToggleHDR(); // Toggle display HDR ON
```

Comment:
This probably works by chance as at re-creating swap chain resets color space to default SDR, then if the intention is render GUI in HDR, `SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020)` should be after `ToggleHDR()`.

Reply:
The GUI is SDR so the color space should always be DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709 after playback, even when the screen is in HDR mode.
Making the GUI HDR at all times would be a separate change to consider, but is not very valuable at this time because there are no HDR elements in the GUI and more GPU resources are consumed by shaders.